### ISSUE TITLE
rustc: not reproducible

### ISSUE THREAD

**Issue by raboof** [NixOS/nixpkgs#229858]
Building this package twice does not produce the bit-by-bit identical result each time, making it harder to detect CI breaches. You can read more about this at https://reproducible-builds.org/ .

Fixing bit-by-bit reproducibility also has additional advantages, such as avoiding hard-to-reproduce bugs, making content-addressed storage more effective and reducing rebuilds in such systems.

### Steps To Reproduce

```
nix-build '<nixpkgs>' -A rustc --check --keep-failed
```

You can use `diffoscope` to analyze the differences in the output of the two builds.

To view the build log of the build that produced the artifact in the binary cache:

```
nix-store --read-log $(nix-instantiate '<nixpkgs>' -A rustc)
```

### Additional context

This unreproducibility seems to be sporadic, but was observed in: https://delroth.net/report-iso_minimal/diff/a0aa598dfe85be4a554c1ed73eb217db8b5f7faeea62b5d8cd5996d8b509550f-80e6fbc77b1d0b38d3dcf1c27ccdea5e1b2906aa2189e24d3a1d24fdfe2406bf.html

Upstream intends to be reproducible: https://github.com/rust-lang/rust/issues/34902

**Comment by ivan770** [comment]
Might be related: https://github.com/rust-lang/rust/issues/112098

**Comment by bachp** [comment]
I tried to reproduce this with:

```
nix-build '<nixpkgs>' -A  rustPackages.rustc-unwrapped --check --keep-failed
```

But I got 3 times the same derivation `/nix/store/praqicbhpa137gq0mb9ygg9c7wyripnk-rustc-1.74.0` accros 2 different machines (1 NixOS, 1 Arch).

I used the following commit on nixpkgs https://github.com/NixOS/nixpkgs/commit/3a360cf8a10d8987c7fb28e4a55239026c0dec98 which was latest master at the time.

Did I do something wrong or is the reproducibility better with 1.74.0?
